stable_baselines/common/vec_env/vec_normalize.py

```diff
@@ -88,48 +89,67 @@ def step_wait(self):
         where 'news' is a boolean vector indicating whether each element is new.
         """
         obs, rews, news, infos = self.venv.step_wait()
-        self.ret = self.ret * self.gamma + rews
         self.old_obs = obs
-        obs = self._normalize_observation(obs)
+        self.old_rews = rews
+
+        if self.norm_obs:
+            if self.training:
+                self.obs_rms.update(obs)
+            obs = self.normalize_obs(obs)
+
         if self.norm_reward:
             if self.training:
-                self.ret_rms.update(self.ret)
-            rews = np.clip(rews / np.sqrt(self.ret_rms.var + self.epsilon), -self.clip_reward, self.clip_reward)
+                self._update_reward(rews)
+            rews = self.normalize_reward(rews)
         self.ret[news] = 0
         return obs, rews, news, infos
 
-    def _normalize_observation(self, obs):
+    def _update_reward(self, rews: np.ndarray) -> None:
+        """Update reward normalizing statistics."""
+        self.ret = self.ret * self.gamma + rews
+        self.ret_rms.update(self.ret)
+
+    def normalize_obs(self, obs: np.ndarray) -> np.ndarray:
         """
-        :param obs: (numpy tensor)
+        Normalize observations using this VecNormalize's observations statistics.
+        Calling this method does not update statistics.
         """
-        if self.norm_obs:
-            if self.training:
-                self.obs_rms.update(obs)
-            obs = np.clip((obs - self.obs_rms.mean) / np.sqrt(self.obs_rms.var + self.epsilon), -self.clip_obs,
-                          self.clip_obs)
-            return obs
-        else:
-            return obs
+        obs = np.clip((obs - self.obs_rms.mean) / np.sqrt(self.obs_rms.var + self.epsilon), -self.clip_obs,
```

Comment:
@AdamGleave Should we add a check like:
```python
if self.norm_obs:
     ...
return obs
```
?

Reply:
In favour of this (and similarly in `normalize_reward`). In that case I think we should also update callees to no longer check `self.norm_obs`.

Reply:
done

Reply:
done